src/c_api.cpp

```diff
@@ -1175,6 +1285,35 @@ int LGBM_BoosterPredictForCSR(BoosterHandle handle,
   API_END();
 }
 
+int LGBM_BoosterPredictForCSRSingleRow(BoosterHandle handle,
+                              const void* indptr,
+                              int indptr_type,
+                              const int32_t* indices,
+                              const void* data,
+                              int data_type,
+                              int64_t nindptr,
+                              int64_t nelem,
+                              int64_t,
+                              int predict_type,
+                              int num_iteration,
+                              const char* parameter,
+                              int64_t* out_len,
+                              double* out_result) {
+  API_BEGIN();
+  auto param = Config::Str2Map(parameter);
```

Comment:
some of this code looks very similar to below, would be nice to refactor to a single common function

Reply:
I agree it looks very much the same, but then again it's mostly boilerplate code that's replicated in other functions too (e.g. LGBM_BoosterPredictForFile). Maybe packaging up the Config creation and translation to OMP would be better. Different PR?

Reply:
ok, ignore my comment then, this is fine.